extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/render/SwitchRenderer.java

```diff
@@ -147,6 +137,44 @@ public boolean canRender(Widget w) {
         return null;
     }
 
+    private void buildButton(Switch w, String lab, String cmd, int maxLabelSize, boolean severalButtons, Item item,
+            State state, StringBuilder buttons) throws RenderException {
+        String button = getSnippet("button");
+
+        String command = cmd;
+        String label = lab;
+
+        if (item instanceof NumberItem && ((NumberItem) item).getDimension() != null) {
+            String unit = getUnitForWidget(w);
+            command = StringUtils.replace(command, UnitUtils.UNIT_PLACEHOLDER, unit);
+            label = StringUtils.replace(label, UnitUtils.UNIT_PLACEHOLDER, unit);
+        }
+
+        if (label != null && maxLabelSize >= 3 && label.length() > maxLabelSize) {
+            label = label.substring(0, maxLabelSize - 3) + "...";
```

Comment:
I wonder if we could save some more space by replacing the three dots by a single unicode character, see https://stackoverflow.com/questions/19997912/how-should-i-write-three-dots. Could you give that a try?

Reply:
Good idea. I will check that when implementing the feature in Classic UI.